client/tracing/src/logging/mod.rs

```diff
@@ -125,21 +125,27 @@ where
 	}
 
 	if directives != "" {
-		// We're not sure if log or tracing is available at this moment, so silently ignore the
-		// parse error.
 		env_filter = parse_user_directives(env_filter, directives)?;
 	}
 
+	if let Some(profiling_targets) = profiling_targets {
+		env_filter = parse_user_directives(env_filter, profiling_targets)?;
+		env_filter = env_filter
+			.add_directive(
+				parse_default_directive("sc_tracing=trace").expect("provided directive is valid")
+			);
+	}
+
 	let max_level_hint = Layer::<FmtSubscriber>::max_level_hint(&env_filter);
 
-	let max_level = max_level.unwrap_or_else(|| match max_level_hint {
+	let max_level = match max_level_hint {
```

Comment:
Can we `use tracing_subscriber::filter::LevelFilter` and clean up some line noise here?